src/wasm/wasm-binary.cpp

```diff
@@ -7616,6 +7568,21 @@ bool WasmBinaryReader::maybeVisitStringNew(Expression*& out, uint32_t code) {
   return true;
 }
 
+bool WasmBinaryReader::maybeVisitStringAsWTF16(Expression*& out,
+                                               uint32_t code) {
+  if (code != BinaryConsts::StringAsWTF16) {
+    return false;
+  }
+  // Parse `string.as_wtf16` as a nop. We do not support this instruction in the
+  // IR, but parsing it as a nop will ensure that the subsequent
+  // `stringview_wtf16.get_codeunit` or `stringview_wtf16.slice` will still
+  // receive the string value. We need to at least accept this in the binary
+  // parser because we emit `string.as_wtf16` as part of the binary encoding of
+  // those other two instructions.
+  out = Builder(wasm).makeNop();
+  return true;
```

Comment:
We no longer parse the stringview types, so it's not possible for the result of the binary `string.as_wtf16` to flow into a local, block result, function result, global, etc. It can only possibly flow directly into `stringview_wtf16.get_codeunit` or `stringview_wtf16.slice`, which we model as taking the `stringref` directly.

Reply:
But how is the stringref input to the `string.as` get hooked up to the parent? That is, the code here seems to replace it with a `nop`, but clearly the parent is skipping that `nop` somehow.

Reply:
The nop is handled just like a nop anywhere else would be: it doesn't change the type of the stack so it's always valid, but it makes the code "stacky" and generally requires inserting a block.

Note that if the binary parser used `IRBuilder`, it would be trivial to just ignore the `string.as_wtf16` rather than parse it as a `nop`. That's what we do in the parser. But as it is, there's no way I could see to both accept the encoding and skip producing an expression for it.